stages/test/test_tar.py

```diff
@@ -145,3 +146,17 @@ def test_tar_transform_no_sh(tmp_path, stage_module, fake_inputs):
     with pytest.raises(subprocess.CalledProcessError) as ex:
         stage_module.main(fake_inputs, tmp_path, options)
     assert "exit status 2" in str(ex.value)
+
+
+@pytest.mark.parametrize('tmp_path_disk_full_size', [5 * 1024])
+@pytest.mark.skipif(os.getuid() != 0, reason="needs root")
+def test_tar_disk_full(stage_module, fake_inputs, tmp_path_disk_full, capfd):
+    options = {
+        "filename": "out.tar",
+    }
+
+    with pytest.raises(subprocess.CalledProcessError) as ex:
+        stage_module.main(fake_inputs, tmp_path_disk_full, options)
+
+    assert ex.value.returncode == 2
+    assert re.match(r".*Wrote only \d+ of \d+ bytes.*", str(capfd.readouterr().err))
```

Comment:
This will fail to match if there is an error message before the one we're trying to match because tar separates them with linebreaks (which `.` doesn't match).

For example, on a system without SELinux support, you get:
```
>       assert re.match(r".*Wrote only \d+ of \d+.*", str(capfd.readouterr().err))
E       AssertionError: assert None
E        +  where None = <function match at 0x762cb7322980>('.*Wrote only \\d+ of \\d+.*', 'tar: SELinux support is not available\ntar: /tmp/pytest-of-root/pytest-13/test_tar_disk_full_5120_0/small-dir/out.tar: Wrote only 8192 of 10240 bytes\ntar: Error is not recoverable: exiting now\n')
E        +    where <function match at 0x762cb7322980> = re.match
E        +    and   'tar: SELinux support is not available\ntar: /tmp/pytest-of-root/pytest-13/test_tar_disk_full_5120_0/small-dir/out.tar: Wrote only 8192 of 10240 bytes\ntar: Error is not recoverable: exiting now\n' = str('tar: SELinux support is not available\ntar: /tmp/pytest-of-root/pytest-13/test_tar_disk_full_5120_0/small-dir/out.tar: Wrote only 8192 of 10240 bytes\ntar: Error is not recoverable: exiting now\n')
E        +      where 'tar: SELinux support is not available\ntar: /tmp/pytest-of-root/pytest-13/test_tar_disk_full_5120_0/small-dir/out.tar: Wrote only 8192 of 10240 bytes\ntar: Error is not recoverable: exiting now\n' = CaptureResult(out='', err='tar: SELinux support is not available\ntar: /tmp/pytest-of-root/pytest-13/test_tar_disk_full_5120_0/small-dir/out.tar: Wrote only 8192 of 10240 bytes\ntar: Error is not recoverable: exiting now\n').err
E        +        where CaptureResult(out='', err='tar: SELinux support is not available\ntar: /tmp/pytest-of-root/pytest-13/test_tar_disk_full_5120_0/small-dir/out.tar: Wrote only 8192 of 10240 bytes\ntar: Error is not recoverable: exiting now\n') = readouterr()
E        +          where readouterr = <_pytest.capture.CaptureFixture object at 0x762cb57e24e0>.readouterr
```

But this could happen in any scenario where a different error is introduced or `tar` changes its error output (for example, it might change to: `Failed to create archive:\nWrote only ...`).

Changing it to `re.search()` should fix the issue.

Reply:
Thanks for the hint!
… please check